File: www/cgi/perl/env.pl
#!/usr/bin/perl

use strict;
use warnings;
use POSIX;
use File::Basename;
use URI::Escape;

my $time = localtime;

my $meth = $ENV{'REQUEST_METHOD'};
if(!defined($meth))
{
    $meth = '------';
}

my $data = '';
if(($meth =~ /POST/))
{
 #   $data=uri_unescape(<STDIN>);
    $data=(<STDIN>);
}

print "Content-type: text/html; charset=utf-8\n\n";
print "<!DOCTYPE html>
<html>
 <head>
  <title>Environment Dumper </title>
  <style>
    body {
        margin-left:50px;
		margin-right:50px;
		background: rgb(60,40,40);
		color: gold;
    }
  </style>
 </head>
 <body>\n";
#  <center>
#   <table border=1>\n";
foreach (sort keys %ENV)
{
    print "   $_ = $ENV{$_}<br>\n";
}
#   </table>
print "
  <p>$data</p>
  <form action=\"env.pl\" method=\"$meth\">
   <input type=\"hidden\" name=\"name\" value=\".-./. .+.!.?.,.~.#.&.>.<.^.\">
   <input type=\"submit\" value=\"Get ENV\">
  </form>
  <hr>
  $time
 </body>
</html>";

File: www/cgi/python/env.py
#!/usr/bin/python3

import os, datetime, sys
import cgi
import codecs

sys.stdout = codecs.getwriter("utf-8")(sys.stdout.detach(), errors='surrogateescape')

print("Content-Type: text/html\n\n")
print("""<!DOCTYPE HTML>
<html>
<head>
<meta charset='utf-8'>
<title>Тест</title>
</head>
<body>
<center>
<table border=1>
""")
#cgi.print_environ()  .decode("utf-8").encode("utf-8")
list_env = os.environ
for i in list_env:
	print('<tr><td>{}</td><td>{}</td></tr>'.format(i, list_env[i]))
now = datetime.datetime.utcnow()
print("""</table>
<form action=\"env.py\" method=\"%s\">
<input type=\"hidden\" name=\"name\" value=\".-./. .+.!.?.,.~.#.&.>.<.^.\">
<p><input type=\"submit\" value=\"Get ENV\"></p>
</form>
</center>
<hr>
%s
</body>
</html>""" %(os.environ['REQUEST_METHOD'], now.strftime("%a, %d %b %Y %H:%M:%S GMT")))
